### ISSUE TITLE
breakpoint not being hit in remote debugging

### ISSUE THREAD

**Issue by jots** [rubyide/vscode-ruby#42]
### Your environment
- `vscode-ruby` version: 0.5.2
- Ruby version: ruby 2.2.2p95
- VS Code version: Version 1.3.0-insider
- Operating System: ubuntu 14.04
- ruby-debug-ide : 0.6.0

config looks like:

``` json
{
   "name": "Listen for rdebug-ide",
   "type": "Ruby",
   "request": "attach",
   "cwd": "${workspaceRoot}",
   "remoteHost": "127.0.0.1",
   "remotePort": "1234",
   "remoteWorkspaceRoot": "${workspaceRoot}"
 }
```

Make sure you have `ruby`, `ruby-debug-ide` and `ruby-debug-basex19` installed before submitting your issue -- thank you !
### Expected behavior

on remote server I:

```
$ rdebug-ide try.rb 
Fast Debugger (ruby-debug-ide 0.6.0, debase 0.2.1, file filtering is supported) listens on 127.0.0.1:1234
```

I have ssh tunnel to remote host on same port.
I set a breakpoint in the code.

I click run in debugger.
expect debugger to stop at breakpoint, It doesn't it runs the program to completion.
### Actual behavior

doesn't stop at breakpoint.

**Comment by codemeow5** [comment]
+1

```
ubuntu@ubuntu:~/workspace/Carnival$ bundle exec rdebug-ide --debug --host 0.0.0.0 --port 1234 --dispatcher-port 1234 main.rb
Fast Debugger (ruby-debug-ide 0.6.0, debase 0.2.1, file filtering is supported) listens on 0.0.0.0:1234
Connected from 192.168.10.9
18992: Starting control thread
18992: Processing in control: break h:\Carnival\main.rb:7
18992: <breakpointAdded no="1" location="h:\Carnival\main.rb:7"/>
18992: Processing in control: break h:\Carnival\main.rb:9
18992: <breakpointAdded no="2" location="h:\Carnival\main.rb:9"/>
18992: Processing in control: break h:\Carnival\main.rb:5
18992: <breakpointAdded no="3" location="h:\Carnival\main.rb:5"/>
18992: Processing in control: break h:\Carnival\main.rb:3
18992: <breakpointAdded no="4" location="h:\Carnival\main.rb:3"/>
18992: Processing in control: start
18992: Starting: running program script
```
and then the program is over.

**Comment by tiansivive** [comment]
I have the same problem. 
I'm using Vagrant and I can successfully launch the debug-ide on the remote VM. Then, after clicking the debug arrow in VScode, it launches my Rails app and everything works fine. However, if I set some breakpoints and try to load the related webpages, it just freezes and the webpage never loads, as if the breakpoint had been hit, except it doesn't. There's no indication of anything happening whatsoever on the vscode debug panel.

**Comment by johnkchiu** [comment]
I just ran into this issue with VSCode and was able to fix it by changing the `remoteWorkspaceRoot` in the launch.json configuration to the path of the Vagrant guest.  By default it has:

```
        {
            "name": "Listen for rdebug-ide",
            "type": "Ruby",
            "request": "attach",
            "cwd": "${workspaceRoot}",
            "remoteHost": "127.0.0.1",
            "remotePort": "1234",
            "remoteWorkspaceRoot": "${workspaceRoot}"
        }
```

Change:
```
"remoteWorkspaceRoot": "/home/vagrant/XXXX"
```

**Comment by Totemika** [comment]
@codemeow5 What should it point to?

**Comment by wingrunr21** [comment]
Closing for issue cleanup. Apologies if this is still an issue. We are working to improve the core extension experience.

**Comment by KevinSnyderCodes** [comment]
@johnkchiu is correct: you need to give `"remoteWorkspaceRoot"` the correct value. The value should be the absolute path where your remote Ruby project lives.

For my Docker image, this was in `/usr/src/app`, so I set `"remoteWorkspaceRoot": "/usr/src/app"` and breakpoints started working.

**Comment by bboyhobbit** [comment]
I finally managed to get this working (with some help!) in the end using the insider build of VS code (1.35.0-insider).  Set up as follows

`        {
            "name": "Rails server",
            "type": "Ruby",
            "request": "launch",
            "cwd": "${workspaceRoot}",
            "showDebuggerOutput": true,
            "program": "${workspaceRoot}/bin/rails",
            "args": [
               "s"
            ]
        }`

I had to uninstall older versions of debase and ruby-debug-ide and update to the latest beta versions.  It now respects breakpoints.  Current Version information:
Ruby: 2.5.1
Rails: 5.2.0
ruby-debug-ide: (0.7.0.beta7)
Debase:   0.2.3.beta5

I did the following:
1. bundle binstubs ruby-debug-ide
2.  bundle binstubs railties
3. bundle binstubs bundler --force
